src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -165,80 +177,65 @@ export default class AmalgamatingBusinesses extends Mixins(CommonMixin) {
   // Local properties
   initialBusinessLookupObject = EmptyBusinessLookup
   businessTableValid = false
+  snackbar = false
+  snackbarText = ''
 
   // Button properties
   isAddingAmalgamatingBusiness = false
   isAddingAmalgamatingForeignBusiness = false
 
-  readonly BusinessLookupServices = BusinessLookupServices
-
-  // Cancel button in "Add an Amalgamating Business" is pressed.
-  addAmalgamatingBusinessCancel (): void {
-    this.isAddingAmalgamatingBusiness = false
-    this.setAmalgamatingBusinessesValid(true)
-  }
-
-  // "Add an Amalgamating Business" button is pressed.
-  onAddBusinessClick (): void {
-    this.isAddingAmalgamatingBusiness = true
-    this.isAddingAmalgamatingForeignBusiness = false
-    this.setAmalgamatingBusinessesValid(false)
-  }
-
-  // "Add an Amalgamating Foreign Business" button is pressed.
-  onAddForeignBusinessClick (): void {
-    this.isAddingAmalgamatingBusiness = false
-    this.isAddingAmalgamatingForeignBusiness = true
-    this.setAmalgamatingBusinessesValid(false)
-  }
-
-  async saveAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
-    let business = null
-
-    // Get the amalgamating business information, mailing address, and email if in LEAR.
-    // Otherwise, return the businesslookup object.
-    const data = await Promise.all([
-      LegalServices.fetchBusinessInfo(businessLookup.identifier),
+  async saveAmalgamatingBusiness (businessLookup: BusinessLookupResultIF): Promise<void> {
+    // Get the auth info, business info, addresses and filings in parallel.
+    // Return data array; if any call failed, that item will be undefined.
+    const data = await Promise.allSettled([
       AuthServices.fetchAuthInfo(businessLookup.identifier),
-      LegalServices.fetchAddresses(businessLookup.identifier)
-    ]).catch((error) => {
-      return error
-    })
+      LegalServices.fetchBusinessInfo(businessLookup.identifier),
+      LegalServices.fetchAddresses(businessLookup.identifier),
+      LegalServices.fetchFilings(businessLookup.identifier)
+    ]).then(results => results.map((result: any) => result.value))
 
-    if (data.length === 3) {
-      business = data[0].data?.business
-      business.businessContact = data[1].contacts[0]
-      business.officeAddress = data[2]
+    // check if all elements are undefined
+    if (!data.some(d => d)) {
+      this.snackbarText = 'Unable to add that business.'
+      this.snackbar = true
+      return
     }
```

Comment:
Karim, what it your experience when a business is in Lear but not affiliated to the current account? I'd like to allow what specifically here and display an error for the other cases.

Reply:
I'm not quite sure if I'm following here Sev. What do you mean?

Reply:
We do 3 or 4 fetches on a business returned from the lookup. What do those fetches return if the business is not affiliated to the current account?

Reply:
Just tested, those fetches return the exact same thing if or not the business is affiliated to the current account.

Reply:
You tested as staff, right?

I tested as BCREG0020 and `0871425` seems to return a 403 for auth info.

Reply:
PS Staff don't need to affiliate so I don't care about that.

Reply:
Weird, I tested as a regular user and made sure to select a business that's not affiliated with that regular user. Sec, trying again..

Reply:
Never mind, you're absolutely right. I don't know why at first try it was fine.

OK, so `fetchBusinessInfo` but the other two don't.

- `fetchAuthInfo`:
  403 (FORBIDDEN)

- `fetchAddresses`:
  401 (UNAUTHORIZED)

Reply:
Theoretically, all 4 should return HTTP 401, but I'm OK with HTTP 403.

I create this bug for the 2 endpoints that allow an unauthorized account from fetching data:
https://app.zenhub.com/workspaces/entities-team-space-6143567664fb320019b81f39/issues/gh/bcgov/entity/18968

Reply:
So I answered my own question here. Resolving.